hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/metadata/SparkHoodieBackedTableMetadataWriter.java

```diff
@@ -144,6 +157,63 @@ protected JavaRDD<HoodieRecord> convertHoodieDataToEngineSpecificData(HoodieData
     return HoodieJavaRDD.getJavaRDD(records);
   }
 
+  @Override
+  protected HoodieData<WriteStatus> convertEngineSpecificDataToHoodieData(JavaRDD<WriteStatus> records) {
+    return HoodieJavaRDD.of(records);
+  }
+
+  @Override
+  protected HoodieData<HoodieRecord> repartitionByMDTFileSlice(HoodieData<HoodieRecord> records, int numPartitions) {
+    return HoodieJavaRDD.of(HoodieJavaRDD.getJavaRDD(records).mapToPair(new PairFunction<HoodieRecord, Pair<String, String>, HoodieRecord>() {
+
+      @Override
+      public Tuple2<Pair<String, String>, HoodieRecord> call(HoodieRecord record) throws Exception {
+        return new Tuple2<>(Pair.of(record.getPartitionPath(), record.getCurrentLocation().getFileId()), record);
+      }
+    }).partitionBy(new Partitioner() {
+      @Override
+      public int numPartitions() {
+        return numPartitions;
+      }
+
+      @Override
+      public int getPartition(Object key) {
+        Pair<String, String> entry = (Pair<String, String>) key;
+        return mapPartitionKeyToSparkPartition(entry.getKey().concat(entry.getValue()), numPartitions);
+      }
+    }).values());
+  }
+
+  @Override
+  public JavaRDD<WriteStatus> streamWriteToMetadataTable(Pair<List<Pair<String, String>>, HoodieData<HoodieRecord>> mdtRecordsHoodieData, String instantTime, boolean initialCall) {
+    JavaRDD<HoodieRecord> mdtRecords = HoodieJavaRDD.getJavaRDD(mdtRecordsHoodieData.getValue());
+
+    if (initialCall) {
```

Comment:
we may not need `initialcall` arg only. 
we can just rely on mdtRecordsHoodieData and if FILES is not present, its the initial call